src/tests/ftest/util/server_utils.py

```diff
@@ -50,292 +47,352 @@ class ServerFailed(Exception):
-        bad_hosts = [
-            node for key in result if key != 1 for node in list(result[key])]
-        kill_server(bad_hosts)
-        raise ServerFailed(
-            "DAOS server processes detected after attempted stop on {}".format(
-                ", ".join([str(result[key]) for key in result if key != 1])))
-
-    # we can also have orphaned ssh processes that started an orted on a
-    # remote node but never get cleaned up when that remote node spontaneiously
-    # reboots
-    subprocess.call(["pkill", "^ssh$"])
-
-
-def kill_server(hosts):
-    """Forcably kill any daos server processes running on the specified hosts.
-
-    Sometimes stop doesn't get everything.  Really whack everything with this.
+        return True
 
-    Args:
-        hosts (list): list of host names where servers are running
-    """
-    kill_cmds = [
-        "pkill '(daos_server|daos_io_server)' --signal INT",
-        "sleep 5",
-        "pkill '(daos_server|daos_io_server)' --signal KILL",
-    ]
-    # Intentionally ignoring the exit status of the command
-    pcmd(hosts, "; ".join(kill_cmds), False, None, None)
+    def stop(self):
+        """Stop the server through the runner."""
+        if self.runner.job.yaml.is_nvme():
+            storage_reset(self._hosts)
+
+        print("Stopping servers")
+        error_list = []
+        try:
+            self.runner.stop()
+        except CommandFailure as error:
+            error_list.append("Error stopping servers: {}".format(error))
+
+        return error_list
+
+    def prep(self):
+        """Prepare the hosts before starting daos server.
+
+        Args:
+            hosts (list): list of host names to prepare
+
+        Raises:
+            ServerFailed: if there is any errors preparing the hosts
+
+        """
+        # Kill any doas servers running on the hosts
+        self.kill()
+
+        # Clean up any files that exist on the hosts
+        self.clean(scm=True)
+
+        # Verify the domain socket directory is present and owned by this user
+        user = getpass.getuser()
+        file_checks = (
+            ("Server", self._hosts, "/var/run/daos_server"),
+        )
+        for host_type, host_list, directory in file_checks:
+            status, nodeset = check_file_exists(host_list, directory, user)
+            if not status:
+                raise ServerFailed(
+                    "{}: {} missing directory {} for user {}.".format(
+                        nodeset, host_type, directory, user))
+
+    def kill(self):
+        """Forcably kill any daos server processes running on hosts.
+
+        Sometimes stop doesn't get everything.  Really whack everything
+        with this.
+
+        """
+        kill_cmds = [
+            "pkill '(daos_server|daos_io_server)' --signal INT",
+            "sleep 5",
+            "pkill '(daos_server|daos_io_server)' --signal KILL",
+        ]
+        # Intentionally ignoring the exit status of the command
```

Comment:
Are you?  Is this comment still valid?  It looks like the code you are replacing was ignoring the exit status but that doesn't seem to be the case any more.

Reply:
You are right, I removed the checks in order to actually ignore the exit status here.

Reply:
Correction: the exit status is still being ignored as `None` is being passed as the `expect_rc` argument to `pcmd`.